internal/controller/supervisorconfig/secretgenerator/secretgenerator.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2020 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package secretgenerator provides a controller that can ensure existence of a generated secret.
+package secretgenerator
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+
+	pinnipedcontroller "go.pinniped.dev/internal/controller"
+	"go.pinniped.dev/internal/controllerlib"
+	"go.pinniped.dev/internal/plog"
+)
+
+const (
+	symmetricKeySecretType    = "secrets.pinniped.dev/symmetric"
```

Comment:
Wondering whether we could use one of the built in Kubernetes `[SecretType](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go#L5773-L5854)`'s for this?

Reply:
My gut says we should stick to our own type so that we have freedom to add data fields to the `Secret` as our feature set grows. We could potentially pose a new `SecretType` upstream if we end up really likely our schema. I don't know of a symmetric key `SecretType`, but perhaps there is one you were thinking of specifically?